pkg/ingester/ingester.go

```diff
@@ -204,6 +207,18 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 		flushQueues:  make([]*util.PriorityQueue, cfg.ConcurrentFlushes, cfg.ConcurrentFlushes),
 	}
 
+	i.flushQueuesDone.Add(cfg.ConcurrentFlushes)
+	for j := 0; j < cfg.ConcurrentFlushes; j++ {
+		i.flushQueues[j] = util.NewPriorityQueue(i.metrics.flushQueueLength)
+		go i.flushLoop(j)
+	}
+
+	if cfg.FlushMode {
+		// Flush and don't update the ring.
+		i.Flush()
+		return i, nil
```

Comment:
Could we initiate the shutdown here?

Reply:
I have been staring at this for a while and I might need some pointers on how cortex can be shut down within ingester. I can only see signals as an option.

Reply:
Hrm, let me see. Can we pass `t.Stop` to the ingester when initiating and then just call that?

Reply:
Will try that! Should work

Reply:
Causes an initialization loop :/

Reply:
Also, I think it's not safe to stop all modules while we are in the process of initing all the modules (additionally, the server is started after all modules are inited).

Reply:
I tried another modification of my earlier attempt - the modules propagate the info of shutting down via a bool instead of cortex explicitly knowing about the ingester job.